flink/src/main/java/org/apache/iceberg/flink/FlinkFilters.java

```diff
@@ -0,0 +1,266 @@
+  private FlinkFilters() {
+  }
+
+  private static final Pattern STARTS_WITH_PATTERN = Pattern.compile("([^%]+)%");
+
+  private static final Map<FunctionDefinition, Operation> FILTERS = ImmutableMap
+      .<FunctionDefinition, Operation>builder()
+      .put(BuiltInFunctionDefinitions.EQUALS, Operation.EQ)
+      .put(BuiltInFunctionDefinitions.NOT_EQUALS, Operation.NOT_EQ)
+      .put(BuiltInFunctionDefinitions.GREATER_THAN, Operation.GT)
+      .put(BuiltInFunctionDefinitions.GREATER_THAN_OR_EQUAL, Operation.GT_EQ)
+      .put(BuiltInFunctionDefinitions.LESS_THAN, Operation.LT)
+      .put(BuiltInFunctionDefinitions.LESS_THAN_OR_EQUAL, Operation.LT_EQ)
+      .put(BuiltInFunctionDefinitions.IN, Operation.IN)
+      .put(BuiltInFunctionDefinitions.IS_NULL, Operation.IS_NULL)
+      .put(BuiltInFunctionDefinitions.IS_NOT_NULL, Operation.NOT_NULL)
+      .put(BuiltInFunctionDefinitions.AND, Operation.AND)
+      .put(BuiltInFunctionDefinitions.OR, Operation.OR)
+      .put(BuiltInFunctionDefinitions.NOT, Operation.NOT)
+      .put(BuiltInFunctionDefinitions.LIKE, Operation.STARTS_WITH)
+      .build();
+
+  /**
+   * convert flink expression to iceberg expression.
+   * <p>
+   * the BETWEEN, NOT_BETWEEN,IN expression will be auto convert by flink. the BETWEEN will be convert to (GT_EQ AND
+   * LT_EQ), the NOT_BETWEEN will be convert to (LT_EQ OR GT_EQ), the IN will be convert to OR, so we do not add the
+   * convert here
+   *
+   * @param flinkExpression the flink expression
+   * @return the iceberg expression
+   */
+  public static Optional<Expression> convert(org.apache.flink.table.expressions.Expression flinkExpression) {
+    if (!(flinkExpression instanceof CallExpression)) {
+      return Optional.empty();
+    }
+
+    CallExpression call = (CallExpression) flinkExpression;
+    Operation op = FILTERS.get(call.getFunctionDefinition());
+    if (op != null) {
+      switch (op) {
+        case IS_NULL:
+          Optional<String> name = toReference(singleton(call, FieldReferenceExpression.class).orElse(null));
+          return name.map(Expressions::isNull);
+
+        case NOT_NULL:
+          Optional<String> nameNotNull = toReference(singleton(call, FieldReferenceExpression.class).orElse(null));
+          return nameNotNull.map(Expressions::notNull);
+
+        case LT:
+          return convertComparisonExpression(Expressions::lessThan, Expressions::greaterThan, call);
+
+        case LT_EQ:
+          return convertComparisonExpression(Expressions::lessThanOrEqual, Expressions::greaterThanOrEqual, call);
+
+        case GT:
+          return convertComparisonExpression(Expressions::greaterThan, Expressions::lessThan, call);
+
+        case GT_EQ:
+          return convertComparisonExpression(Expressions::greaterThanOrEqual, Expressions::lessThanOrEqual, call);
+
+        case EQ:
+          return handleNaN(Expressions::equal, call);
+
+        case NOT_EQ:
+          return handleNaN(Expressions::notEqual, call);
```

Comment:
I think there's a bug here.   Assume the case:  `a != NaN`,   the `handleNaN` will return an iceberg expression:  `Expressions.isNaN(name)`.  That's incorrect ?

Reply:
Pls add an unit test for this if possible.

Reply:
> I think there's a bug here. Assume the case: `a != NaN`, the `handleNaN` will return an iceberg expression: `Expressions.isNaN(name)`. That's incorrect ?

It was my negligence, forgot to deal with `!=`

Reply:
> Pls add an unit test for this if possible.

we can construct `NaN` type by `float('NaN')` in spark and hive sql,
but I lookup the flink [doc](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/types.html),It does not seem to support `Infinity` an `NaN` now